### ISSUE TITLE
Confusing output: "Verified module" even when there were errors

### ISSUE THREAD

**Issue by fmrl** [FStarLang/FStar#169]
The following code is an attempt to create a version of `nth` whose inputs are guarded with verification conditions:

```
module Nth

   // from list.fst
   val length: list 'a -> Tot nat
   let rec length = function
     | [] -> 0
     | _::tl -> 1 + length tl
   let lengthT = length

   val tot_nth : l:list 'a -> n:nat -> Tot (a:'a{0 < (length l) && n < (length l)})
   let rec tot_nth l n =
      match l with
         | hd::tl ->
            if n = 0 then
               hd
            else
               tot_nth tl (n - 1)
```

The code verifies but does not compile:

```
> fstar --verify "nth.fst"
nth.fst(12,6-18,0): Patterns are incomplete
Verified module: Prims
Verified module: Nth
Error: 1 errors were reported (see above)
> exit status: 256
```

My thinking is that `tot_nth` should require no base case because the dependent type `'a{0 < (length l) && n < (length l)}` forbids this.

Am I mistaken? or is there something I'm missing to satisfy the compiler?

**Comment by catalin-hritcu** [comment]
This does not verify, because it's wrong. The refinement should be on the argument, not on the result:
```
   val tot_nth : l:(list 'a) -> n:nat{0 < (length l) && n < (length l)} -> Tot ('a)
```

**Comment by fmrl** [comment]
I understand. Thanks for the diagnosis.

Am I to understand that the message `Verified module: Nth` does not mean it verified successfully-- that it instead means that an attempt to verify the module completed?

**Comment by catalin-hritcu** [comment]
Indeed. We might want to make the message more explicit in case verification had errors.

**Comment by catalin-hritcu** [comment]
Fixed in 38a9c5a
